components/tools/OmeroJava/test/integration/thumbnail/SkipThumbnailsPermissionsTest.java

```diff
@@ -0,0 +1,640 @@
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Tests utilizing single thumbnail loading APIs by users other than the
+ * owner of the image. In each test an image is imported into OMERO without
+ * thumbnails (--skip thumbnails).
+ * @author Riad Gozim &nbsp;&nbsp;&nbsp;&nbsp; <a
+ * href="mailto:r.gozim@dundee.ac.uk">r.gozim@dundee.ac.uk</a>
+ * @since 5.4.10
+ */
+@SuppressWarnings("Duplicates")
+public class SkipThumbnailsPermissionsTest extends AbstractServerImportTest {
+
+    /* Total wait time will be WAITS * INTERVAL milliseconds */
+    /**
+     * Maximum number of intervals to wait for pyramid
+     **/
+    private static final int WAITS = 500;
+
+    /**
+     * Wait time in milliseconds
+     **/
+    private static final long INTERVAL = 1000L;
+
+    /**
+     * The collection of files that have to be deleted.
+     */
+    private List<File> files;
+
+    @BeforeMethod
+    protected void startup() throws Throwable {
+        files = new ArrayList<>();
+        createImporter();
+    }
+
+    @AfterMethod
+    public void cleanup() {
+        for (File file : files) {
+            file.delete();
+        }
+        files.clear();
+    }
+
+    @Test(dataProvider = "permissions")
+    public void testGetThumbnail(String permissions, boolean isAdmin,
+                                 boolean isGroupOwner) throws Throwable {
+        // Create two users in same group
+        EventContext user1 = newUserAndGroup(permissions);
+        loginUser(user1);
+
+        ImportConfig config = new ImportConfig();
+        config.doThumbnails.set(false); // skip thumbnails
+
+        // Import image without thumbnails
+        Pixels pixels = importFile(config);
+
+        // View image as user 1
+        ThumbnailStorePrx svc = factory.createThumbnailStore();
+        Utils.setThumbnailStoreToPixels(svc, pixels.getId().getValue());
+        Utils.getThumbnail(svc);
+        svc.close();
+
+        // Create new user in group and login as that user
+        EventContext user2 = newUserInGroup(user1, isGroupOwner);
+
+        // If user is an admin, add them to the system group
+        if (isAdmin) {
+            ExperimenterGroup systemGroup = new ExperimenterGroupI(iAdmin.getSecurityRoles().systemGroupId, false);
+            addUsers(systemGroup, Collections.singletonList(user2.userId), false);
+        }
+        disconnect();
+        // Login as user2
+        loginUser(user2);
+
+        // Try to load image
+        svc = factory.createThumbnailStore();
+        try {
+            Utils.setThumbnailStoreToPixels(svc, pixels.getId().getValue());
+            Utils.getThumbnail(svc);
+        } catch (omero.ResourceError | omero.ReadOnlyGroupSecurityViolation e) {
+            // With permission rw----, the image is private to user 1.
+            // Expect this to fail for user 2.
+            if (!permissions.equalsIgnoreCase("rw----")) {
```

Comment:
Why are all of these ignoring case?

Reply:
b/c in that case the operation is not permitted and the error is expected. This could be replaced by success message

Reply:
I'm completely lost, sorry. The error would make a simple `equals` insufficient?